_posts/2022-04-11-1.15-announcement.md

```diff
@@ -0,0 +1,375 @@
+different topology. This dynamic setting makes guaranteeing snapshot compatibility 
+really hard across Flink versions. In the efforts of 1.15, the community put the focus 
+on keeping the same query up and running even after upgrades. At the core of SQL 
+upgrades are JSON plans 
+([for now we have only documentation in our JavaDoc, while we are still working on updating the documentation](https://nightlies.apache.org/flink/flink-docs-release-1.15/api/java/org/apache/flink/table/api/CompiledPlan.html)), 
+they have been introduced for 
+internal use already in previous releases and will now be exposed. Both the Table API 
+and SQL will provide a way to compile and execute a plan which guarantees the same 
+topology throughout versions. This feature will be released as an experimental MVP. 
+Users who want to give it a try already can create a JSON plan that can then be used 
+to restore a Flink Job based on the old operator structure. The first real upgrade 
+will then happen in Flink 1.16.
+
+
+### Changelog state backend
+
+From Flink 1.15, we have the MVP feature of 
+[changelog state](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/ops/state/state_backends/#enabling-changelog)
+backend, which aims at 
+making checkpoint intervals shorter with following advantages:
+
+1. Less work on recovery: The more frequently a checkpoint is taken, the fewer events 
+   need to be re-processed after recovery.
+2. Lower latency for transactional sinks: Transactional sinks commit on checkpoints, 
+   so faster checkpoints mean more frequent commits.
+3. More predictable checkpoint intervals: Currently the length of the checkpoint mainly
+   depends on the size of the artifacts that need to be persisted on the checkpoint 
+   storage, by keeping the size small it becomes more predictable.
+
+This work introduced in Flink 1.15 helps achieve the above advantages by continuously 
+persisting state changes on a non-volatile storage, while performing materialization 
+in the background.
+
+
+### Adaptive batch scheduler
+
+In 1.15, we introduced a new scheduler to Apache Flink: the 
+[Adaptive Batch Scheduler](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/elastic_scaling/#adaptive-batch-scheduler). 
+The new scheduler can automatically decide parallelisms of job vertices for batch jobs, 
+according to the size of data volume each vertex needs to process.
+
+
+### Other things worth mentioning
+
+There have been improvements to the application mode. Jobs that should take a savepoint 
+after they are completed, can now guarantee they do so 
+([see execution.shutdown-on-application-finish](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/config/#execution-shutdown-on-application-finish)). 
+Recovery and clean up of jobs running in application mode have been improved. The local 
+state can be persisted in the working directory to improve the experience of recovering 
+from local storage.
```

Comment:
```suggestion
### Improvements to application mode

When running Flink in [application mode](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/deployment/overview/), it can now be guaranteed that jobs will take a savepoint after they are completed if they have been configured to do so
([see execution.shutdown-on-application-finish](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/config/#execution-shutdown-on-application-finish)). 

The recovery and clean up of jobs running in application mode has also been improved. The local 
state can be persisted in the working directory, which makes recovering 
from local storage easier.
```